refinery/core/views.py

```diff
@@ -1094,7 +1095,12 @@ def delete(self, request, uuid):
                 content="User {} is not authenticated".format(request.user))
         else:
             try:
-                dataset_deleted = DataSet.objects.get(uuid=uuid).delete()
+                with transaction.atomic():
+                    dataset_deleted = DataSet.objects.get(uuid=uuid).delete()
```

Comment:
This should be split up into get and delete for readability and to handle `DataSet.DoesNotExist` exception.